### PR TITLE
Ea/disappearing qa fix

### PR DESCRIPTION
**Pull request by awisema**

### CODE REVIEW

File: app/views/health/sdh_case_management_notes/edit.haml [hunk 1]
```diff
@@ -18,3 +19,40 @@
         }
       });
     });
+
+    // Show type in box for other values
+    $('body').on('change', '.string-collection-with-other__control', function(e) {
```
Comment (eanders) [inline app/views/health/sdh_case_management_notes/edit.haml:23]: This is the same as line 44, so I think you can collapse the two into one `on` hook.

Comment (awisema) [reply app/views/health/sdh_case_management_notes/edit.haml:23]: Yes, it is the same, and in the original code they were together -- it just seemed clearer to me, to have two smaller methods that handle the two different cases. If you would prefer one `on` hook, I'll move it into 2 functions called from a single hook?

Comment (eanders) [reply app/views/health/sdh_case_management_notes/edit.haml:23]: I think one hook is better, I worry about duplicate hooks touching the same stuff, it means potentially calling the same thing twice, which can cause issues down the line.

File: app/views/health/qualifying_activities/destroy.js.coffee [hunk 2]
```diff
@@ -0,0 +1,4 @@
+html = "<%= j render('health/sdh_case_management_notes/activities', qa: @qa) %>"
+$('.modal:visible .close').trigger('click')
+$container = $('.jActivityList')
+$container.replaceWith(html)
```
Comment (eanders) [inline app/views/health/qualifying_activities/destroy.js.coffee:4]: replace with `<%= j render 'create' %>` (but we need to test)  the JS code is the same.

File: app/views/health/qualifying_activities/update.js.coffee [hunk 3]
```diff
@@ -0,0 +1,4 @@
+html = "<%= j render('health/sdh_case_management_notes/activities', qa: @qa) %>"
+$('.modal:visible .close').trigger('click')
+$container = $('.jActivityList')
+$container.replaceWith(html)
```
Comment (eanders) [inline app/views/health/qualifying_activities/update.js.coffee:4]: replace with `<%= j render 'create' %>` (but we need to test)  the JS code is the same.

File: app/models/health/sdh_case_management_note.rb [hunk 4]
```diff
@@ -194,13 +194,13 @@ def user_can_edit?(current_user)
...
-        {id: :additional_questions, title: 'Additional Questions'}
-      ]
-    end
+    # def display_note_form_sections
+    #   [
+    #     {id: :basic_info, title: 'Note Details'},
+    #     {id: :activities, title: 'Qualifying Activities'},
+    #     {id: :additional_questions, title: 'Additional Questions'}
+    #   ]
+    # end
```
Comment (eanders) [inline app/models/health/sdh_case_management_note.rb]: Remove

File: app/controllers/health/qualifying_activities_controller.rb [hunk 5]
```diff
@@ -13,9 +13,57 @@ class QualifyingActivitiesController < IndividualPatientController
...
+    end
+
+    def update
+      unversioned_params = qa_params.to_h
+      @qa = @patient.qualifying_activities.build(date_of_activity: unversioned_params[:date_of_activity], user: current_user, user_full_name: current_user.name)
+      unversioned_params.delete_if do |k, _|
+        ! k.in?(@qa.qa_version.class.versioned_attribute_names)
+      end
+
+      if @qa.update(unversioned_params)
```
Comment (eanders) [inline app/controllers/health/qualifying_activities_controller.rb:56]: `@qa.delay.maintain_cached_values`

File: app/controllers/health/qualifying_activities_controller.rb [hunk 6]
```diff
@@ -13,9 +13,57 @@ class QualifyingActivitiesController < IndividualPatientController
...
+    end
+
+    def create
+      unversioned_params = qa_params.to_h
+      @qa = @patient.qualifying_activities.build(date_of_activity: unversioned_params[:date_of_activity], user: current_user, user_full_name: current_user.name)
+      unversioned_params.delete_if do |k, _|
+        ! k.in?(@qa.qa_version.class.versioned_attribute_names)
+      end
+
+      if @qa.update(unversioned_params)
```
Comment (eanders) [inline app/controllers/health/qualifying_activities_controller.rb:38]: `@qa.delay.maintain_cached_values`

File: app/controllers/health/sdh_case_management_notes_controller.rb [hunk 7]
```diff
@@ -37,7 +37,7 @@ def new
     end
 
     def edit
-      @note.activities.build(date_of_activity: Date.current) if @note.activities.size.zero? # Add a qa if there isn't at least one
+      # @note.activities.build(date_of_activity: Date.current) if @note.activities.size.zero? # Add a qa if there isn't at least one
```
Comment (eanders) [inline app/controllers/health/sdh_case_management_notes_controller.rb]: remove comments in this file

File: app/views/health/qualifying_activities/destroy.js.coffee (hunk 2 above)
Comment (awisema) [reply app/views/health/qualifying_activities/destroy.js.coffee:4]: I couldn't get this to work in the time allotted  -- tabling for future exploration.

**Pull request merged**